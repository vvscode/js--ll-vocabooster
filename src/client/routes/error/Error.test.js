import React from 'react';
import { shallow } from 'enzyme';
import Component from './Error.component';

describe('Error', () => {
  let props;
  let getShallowComponent;

  beforeEach(() => {
    props = {
      error: null,
    };

    getShallowComponent = () => shallow(<Component {...props} />);
  });

  describe('render', () => {
    it('should render with no error', () => {
      expect(getShallowComponent()).toMatchSnapshot();
    });

    it('should render with error', () => {
      props.error = {
        name: 'Some error',
        message: 'Some message',
        stack: 'Some stack',
      };
      expect(getShallowComponent()).toMatchSnapshot();
    });
  });
});
