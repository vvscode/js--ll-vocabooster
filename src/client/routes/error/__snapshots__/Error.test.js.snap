// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error render should render with error 1`] = `
<div
  className="Error__root"
>
  <h1
    className="Error__title"
  >
    Some error
  </h1>
  <pre
    className="Error__stack"
  >
    Some stack
  </pre>
</div>
`;

exports[`Error render should render with no error 1`] = `
<div
  className="Error__root"
>
  <h1
    className="Error__title"
  >
    Error
  </h1>
  <p
    className="Error__message"
  >
    Sorry, a critical error occurred on this page.
  </p>
</div>
`;
